File: backend/app/api/routes/assets.py
from uuid import UUID

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.db.database import get_db
from app.schemas.asset import AssetCreate, AssetResponse
from app.services.asset_service import AssetService

router = APIRouter(
    prefix="/assets",
    tags=["Assets"]
)


@router.get("/", response_model=list[AssetResponse])
def get_assets(db: Session = Depends(get_db)):
    return AssetService.get_assets(db)


@router.get("/{asset_id}", response_model=AssetResponse)
def get_asset(asset_id: UUID, db: Session = Depends(get_db)):
    return AssetService.get_asset(db, asset_id)


@router.post("/", response_model=AssetResponse)
def create_asset(asset: AssetCreate, db: Session = Depends(get_db)):
    return AssetService.create_asset(db, asset)

File: frontend/src/services/assets.ts
import api from "./api";

export const getAssets = async () => {
    const response = await api.get("/assets/");
    return response.data;
};

export const createAsset = async (asset: any) => {
    const response = await api.post("/assets/", asset);
    return response.data;
};

export const updateAsset = async (id: string, asset: any) => {
    const response = await api.put(`/assets/${id}`, asset);
    return response.data;
};

export const deleteAsset = async (id: string) => {
    const response = await api.delete(`/assets/${id}`);
    return response.data;
};